lib/phoenix/endpoint/supervisor.ex

```diff
@@ -327,19 +329,26 @@ defmodule Phoenix.Endpoint.Supervisor do
   """
   @invalid_local_url_chars ["\\"]
 
-  def static_path(_endpoint, "//" <> _ = path) do
+  def static_path(endpoint, path) do
+    case static_lookup(endpoint, path) do
+      {:nocache, {path, _}} -> {:nocache, path}
+      other -> other
+    end
+  end
+
+  def static_lookup(_endpoint, "//" <> _ = path) do
     raise_invalid_path(path)
   end
 
-  def static_path(_endpoint, "/" <> _ = path) do
+  def static_lookup(_endpoint, "/" <> _ = path) do
     if String.contains?(path, @invalid_local_url_chars) do
       raise ArgumentError, "unsafe characters detected for path #{inspect path}"
     else
-      {:nocache, path}
+      {:nocache, {path, ""}}
```

Comment:
If we don't have integrity, I would prefer to return `nil`. I left a similar comment below. :)